webapi/src/main/scala/org/knora/webapi/responders/v2/ResourcesResponderV2.scala

```diff
@@ -101,34 +132,129 @@ class ResourcesResponderV2 extends ResponderWithStandoffV2 {
-                (resIri: IRI) =>
-                    ConstructResponseUtilV2.createFullResourceResponse(resIri, queryResultsSeparated(resIri), mappings = Map.empty[IRI, MappingAndXSLTransformation])
-            }.toVector
+            // convert standoff assertions to standoff tags
+            standoffTags: Vector[StandoffTagV2] = StandoffTagUtilV2.createStandoffTagsV2FromSparqlResults(teiMapping.standoffEntities, valueObject.standoff)
+
+            // create XML from standoff (temporary XML) that is going to be converted to TEI/XML
+            tmpXml = StandoffTagUtilV2.convertStandoffTagV2ToXML(valueObject.assertions(KnoraBase.ValueHasString), standoffTags, teiMapping.mapping)
+
+            teiXSLTFile: File = new File("src/main/resources/standoffToTEI.xsl")
+
+            _ = if (!teiXSLTFile.canRead) throw NotFoundException("Cannot find XSL transformation for TEI: 'src/main/resources/standoffToTEI.xsl'")
+
+            // apply XSL transformation to temporary XML to create the TEI/XML body
+            xslt: String = FileUtils.readFileToString(teiXSLTFile, "UTF-8")
+
+            teiXMLBody = XMLUtil.applyXSLTransformation(tmpXml, xslt)
+
+            // header
+
+            // get all the properties but the property representing the text for the body
+            headerProps: Map[IRI, Seq[ConstructResponseUtilV2.ValueRdfData]] = queryResultsSeparated(resourceIri).valuePropertyAssertions - textProperty.toString
+
+            // collect Iris of referred resources
+            referredResourceIris = headerProps.values.flatten.foldLeft(Seq.empty[IRI]) {
+
+                (referredResIris: Seq[IRI], valObj: ConstructResponseUtilV2.ValueRdfData) =>
+
+                    if (valObj.valueObjectClass == OntologyConstants.KnoraBase.LinkValue && !valueObject.incomingLink) {
+                        val refResIRI: IRI = valObj.assertions(OntologyConstants.Rdf.Object)
+
+                        referredResIris :+ refResIRI
+                    } else {
+                        referredResIris
+                    }
+
+            }
+
+            headerInfos = queryResultsSeparated(resourceIri).copy(
+                valuePropertyAssertions = headerProps
+            )
+
+            headerResource: ReadResourceV2 = ConstructResponseUtilV2.createFullResourceResponse(resourceIri, headerInfos, mappings = Map.empty[IRI, MappingAndXSLTransformation])
+
+            headerJSONLD = ReadResourcesSequenceV2(1, Vector(headerResource)).toJsonLDDocument(ApiV2WithValueObjects, settings)
+
+            rdfParser: RDFParser = Rio.createParser(RDFFormat.JSONLD)
+            stringReader = new StringReader(headerJSONLD.toCompactString)
+            stringWriter = new StringWriter()
+
+            rdfWriter: RDFWriter = new RDFXMLPrettyWriter(stringWriter)
+
+            _ = rdfParser.setRDFHandler(rdfWriter)
+            _ = rdfParser.parse(stringReader, "")
+
+            teiHeader = stringWriter.toString
+
+            _ = println(teiHeader)
```

Comment:
@benjamingeer Crazy shit!

```xml
 <?xml version="1.0" encoding="UTF-8"?>
 <rdf:RDF
 	xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
 	xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
 	xmlns:knora-api="http://api.knora.org/ontology/knora-api/v2#"
 	xmlns:anything="http://0.0.0.0:3333/ontology/0001/anything/v2#">
 <anything:Thing rdf:about="http://rdfh.ch/0001/thing_with_richtext_with_markup">
 	<anything:hasOtherThingValue rdf:resource="http://rdfh.ch/0001/thing_with_richtext_with_markup/values/h7Ni5PXGRtSO0_KD_y9OzA"/>
 	<anything:hasText rdf:resource="http://rdfh.ch/0001/thing_with_richtext_with_markup/values/ExkRf8XmSFWr8cwNe9pMEA"/>
 	<rdfs:label rdf:datatype="http://www.w3.org/2001/XMLSchema#string">test thing with markup</rdfs:label>
 </anything:Thing>
 <knora-api:LinkValue rdf:about="http://rdfh.ch/0001/thing_with_richtext_with_markup/values/h7Ni5PXGRtSO0_KD_y9OzA">
 	<knora-api:linkValueHasTarget>
 		<anything:Thing rdf:about="http://rdfh.ch/0001/0C-0L1kORryKzJAJxxRyRQ">
 			<rdfs:label rdf:datatype="http://www.w3.org/2001/XMLSchema#string">Sierra</rdfs:label>
 		</anything:Thing>
 	</knora-api:linkValueHasTarget>
 </knora-api:LinkValue>
 <knora-api:TextValue rdf:about="http://rdfh.ch/0001/thing_with_richtext_with_markup/values/ExkRf8XmSFWr8cwNe9pMEA">
 	<knora-api:valueAsString rdf:datatype="http://www.w3.org/2001/XMLSchema#string">test metadata</knora-api:valueAsString>
 </knora-api:TextValue>
```

If we can get the `ReadResourceV2` via gravsearch and specify which properties we want to be returned for referred resources, we can say that the resource points to another object with a certain property. E.g., the author of a letter who is identified by a IAF/GND.

Reply:
Tomorrow I'll work on making the Gravsearch query possible.